### ISSUE TITLE
UndefinedInterfaceMethod for `Cache::driver()->flexible()`: narrow `CacheManager` returns to concrete `Illuminate\Cache\Repository`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1230]
**Describe the bug**

`Cache::driver()->flexible(...)` (and the same call via `Cache::store()` or an injected `CacheManager`) is reported as `UndefinedInterfaceMethod`:

```
ERROR: UndefinedInterfaceMethod
Method Illuminate\Contracts\Cache\Repository::flexible does not exist (see https://psalm.dev/181)
```

Root cause: `flexible()` (added in Laravel 11.23, SWR caching) is declared only on the concrete `Illuminate\Cache\Repository`, not on the `Illuminate\Contracts\Cache\Repository` interface. `CacheManager::store()` / `driver()` / `memo()` docblock the contract as their return type, so Psalm resolves the call against the interface and fails.

At runtime these methods always return the concrete `Illuminate\Cache\Repository`: every store, including custom `Cache::extend()` drivers, is wrapped by `CacheManager::repository()`. Narrowing the manager's return type to the concrete class is therefore sound and gives access to the full concrete surface (`flexible()`, `tags()`, Macroable) without lying about the interface.

**Proposed fix**

Add a `stubs/common/Cache/CacheManager.phpstub` overriding the return type of `store()`, `driver()` and `memo()` to `\Illuminate\Cache\Repository`. The `Cache` facade path already resolves to `CacheManager` via `FacadeMapProvider`, so the facade call site is fixed by the same stub.

Alternatives considered and rejected:
- Adding `@method flexible(...)` to a contract stub: lies about the interface; a custom `Contracts\Cache\Repository` implementation genuinely lacks the method (false negative).
- Upstream `laravel/framework` change: adding `flexible()` to the contract is a BC break; a docblock-only `@return` narrowing on `CacheManager` may be declined because `Cache::swap()` / container rebinding can inject any contract implementation.

Stub-authoring caveats (per `docs/contributing/`): copy the `CacheManager` class header verbatim (a bare re-declaration strips implements/traits), and keep the version gate in mind: `flexible()` exists since Laravel 11.23, so the `Repository::flexible()` signature itself is fine in `stubs/common/` only if the plugin's supported Laravel floor is >= 11.23; otherwise it belongs in a version dir.

Related: #701 typed these calls down to `Illuminate\Contracts\Cache\Repository`, which is exactly the level that misses concrete-only methods like `flexible()`.

**Code to reproduce**

```php
<?php

use Illuminate\Support\Facades\Cache;

/** @return array<string> */
function loadThings(): array
{
    return Cache::driver('file')->flexible('things', [5 * 60, 60 * 60], static function (): array {
        return ['a', 'b'];
    });
}
```

**Impacted Versions**

```text
laravel/framework    v12.62.0
psalm/plugin-laravel dev-master
vimeo/psalm          7.0.0-beta19
```

**Psalm config**

```xml
<?xml version="1.0"?>
<psalm errorLevel="4" xmlns="https://getpsalm.org/schema/config">
    <projectFiles>
        <directory name="app" />
    </projectFiles>
    <plugins>
        <pluginClass class="Psalm\LaravelPlugin\Plugin"/>
    </plugins>
</psalm>
```

**Additional context**

The `memo()` driver wraps stores in `MemoizedStore` but still returns the concrete `Illuminate\Cache\Repository`, so it takes the same narrowing.
